RadixWallet/Features/GatewaySettingsFeature/Children/AddNewGateway/AddNewGateway+Reducer.swift

```diff
@@ -75,8 +76,13 @@ public struct AddNewGateway: Sendable, FeatureReducer {
 		case let .textFieldChanged(inputtedURL):
 			state.inputtedURL = inputtedURL
 			state.errorText = nil
-			let url = URL(string: inputtedURL)
-			state.addGatewayButtonState = url != nil ? .enabled : .disabled
+			if let url = URL(string: state.inputtedURL)?.httpsURL, let ffiUrl = try? FfiUrl(urlPath: url.absoluteString) {
```

Comment:
whats the reason for having `httpsURL`?

Reply:
We are now building the `URL` & `FfiUrl` only once, and saving it on the state (rather than building it twice as it was done before). 
Then, we need to manually append `https://` if it isn't present to allow users adding just `hammunet-gateway.radixdlt.com`, and not having to add the https prefix. 

This logic was already present before [here](https://github.com/radixdlt/babylon-wallet-ios/pull/1266/files/55a14febc337e0e53806de1ed999dfd9085c98bf#diff-1546cac7f31711a249c112e33ac628d43301b6a9fd82f97560e25a7d95f0f17cL59)